dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/resourcePolicy/ResourcePolicyActionAddOrReplaceOperation.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository.patch.operation.resourcePolicy;
+
+import static org.dspace.app.rest.repository.patch.operation.resourcePolicy.ResourcePolicyUtils.OPERATION_PATH_ACTION;
+
+import org.dspace.app.rest.exception.DSpaceBadRequestException;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.patch.Operation;
+import org.dspace.app.rest.repository.patch.operation.PatchOperation;
+import org.dspace.authorize.ResourcePolicy;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation for ResourcePolicy action ADD or REPLACE patch.
+ *
+ * Example: <code>
+ * curl -X PATCH http://${dspace.server.url}/api/authz/resourcepolicies/<:id-resourcepolicy> -H "
+ * Content-Type: application/json" -d '[{ "op": "replace", "path": "
+ * /action", "value": 2]'
+ * </code>
+ *
+ * @author Emanuele Ballarini (emanuele.ballarini@4science.com)
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+ */
+@Component
+public class ResourcePolicyActionAddOrReplaceOperation<R> extends PatchOperation<R> {
+
+    @Override
+    public R perform(Context context, R resource, Operation operation) {
+        checkOperationValue(operation.getValue());
+        if (this.supports(resource, operation)) {
+            ResourcePolicy resourcePolicy = (ResourcePolicy) resource;
+            this.replace(resourcePolicy, operation);
+            return resource;
+        } else {
+            throw new DSpaceBadRequestException(this.getClass() + " does not support this operation");
+        }
+    }
+
+    /**
+     * Performs the actual add or replace action of resourcePolicy operation. Both
+     * actions are allowed since the starting value of action is a defined int.
+     * 
+     * @param resourcePolicy resourcePolicy being patched
+     * @param operation      patch operation
+     */
+    private void replace(ResourcePolicy resourcePolicy, Operation operation) {
+        int action = (int) operation.getValue();
```

Comment:
This line throws a `ClassCastException` for me anytime I try to change the `action` of an existing Policy. 

For example, I tried to edit a Policy which specified `READ` and change it to `WRITE`.   I received a 500 response, and got this error in the backend logs:
```
2022-05-18 19:40:18,210 ERROR a14cfb2c-4e21-445c-bdb9-b1f53afe8843 475691ec-6638-4c9b-b96a-7e7cf7572654 org.dspace.app.r
est.exception.DSpaceApiExceptionControllerAdvice @ An exception has occurred (status:500)
java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Integer (java.lang.String and jav
a.lang.Integer are in module java.base of loader 'bootstrap')
        at org.dspace.app.rest.repository.patch.operation.resourcePolicy.ResourcePolicyActionAddOrReplaceOperation.repla
ce(ResourcePolicyActionAddOrReplaceOperation.java:56) ~[classes/:7.3-SNAPSHOT]
        at org.dspace.app.rest.repository.patch.operation.resourcePolicy.ResourcePolicyActionAddOrReplaceOperation.perfo
rm(ResourcePolicyActionAddOrReplaceOperation.java:41) ~[classes/:7.3-SNAPSHOT]
        at org.dspace.app.rest.repository.patch.ResourcePatch.performPatchOperation(ResourcePatch.java:58) ~[classes/:7.
3-SNAPSHOT]
        at org.dspace.app.rest.repository.patch.ResourcePatch.patch(ResourcePatch.java:42) ~[classes/:7.3-SNAPSHOT]
        at org.dspace.app.rest.repository.ResourcePolicyRestRepository.patch(ResourcePolicyRestRepository.java:320) ~[cl
asses/:7.3-SNAPSHOT]
```

Reply:
Thanks for the report, we'll take it in charge and work on it in the next few days.